### PR TITLE
Change "toolchain" to "python"

### PR DESCRIPTION
**Pull request by zanieb**
Whew this is a lot.

The user-facing changes are:

- `uv toolchain` to `uv python` e.g. `uv python find`, `uv python install`, ...
- `UV_TOOLCHAIN_DIR` to` UV_PYTHON_INSTALL_DIR`
- `<UV_STATE_DIR>/toolchains` to `<UV_STATE_DIR>/python` (with [automatic migration](https://github.com/astral-sh/uv/pull/4735/files#r1663029330))
- User-facing messages no longer refer to toolchains, instead using "Python", "Python versions" or "Python installations"

The internal changes are:

- `uv-toolchain` crate to `uv-python`
- `Toolchain` no longer referenced in type names
- Dropped unused `SystemPython` type (previously replaced)
- Clarified the type names for "managed Python installations"
- (more little things)

### CODE REVIEW

File: crates/uv-python/src/managed.rs [hunk 1]
```diff
@@ -74,33 +74,48 @@ impl InstalledToolchains {
...
+                .is_some_and(|parent| parent.join("toolchains").exists())
+        {
+            let deprecated = root.parent().unwrap().join("toolchains");
+            // Move the deprecated directory to the new location.
+            fs::rename(&deprecated, root)?;
+            // Create a link or junction to at the old location
+            uv_fs::replace_symlink(root, &deprecated)?;
+        } else {
+            fs::create_dir_all(root)?;
+        }
```
Comment (zanieb) [inline crates/uv-python/src/managed.rs:116]: This blurb is notable

**Review by charliermarsh — APPROVED** [review]

**Comment by charliermarsh** [comment]
My only complaint is that `uv python` feels like I'd go `uv python foo.py` (to run Python itself), but I don't have a better suggestion.

**Comment by zanieb** [comment]
Yeah I agree that's a little confusing and my primary hesitation. What if we provide a `uv python run` command though with an optional `python` shim on the PATH that invokes it?

**Comment by samypr100** [comment]
Would there be a short-hand for this similar to uvx? Say `uvp` or `uv py`?

**Comment by zanieb** [comment]
I think similar to `uvx` mapping to `uv tool run` (but a little different in implementation) we'd allow opt-in installation of `python` (or another name) that maps to `uv python run`.

**Pull request merged**